plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/DragDropWithUpdateNodeTest.java

```diff
@@ -0,0 +1,102 @@
+
+    private static final String REPRESENTATION_NAME = "new " + REPRESENTATION_DESCRIPTION_NAME;
+
+    @Override
+    protected void onSetUpBeforeClosingWelcomePage() throws Exception {
+        copyFileToTestProject(Activator.PLUGIN_ID, PATH, SEMANTIC_MODEL, REPRESENTATION_MODEL, MODELER);
+    }
+
+    @Override
+    protected void onSetUpAfterOpeningDesignerPerspective() throws Exception {
+        changeSiriusPreference(SiriusPreferencesKeys.PREF_AUTO_REFRESH.name(), true);
+        sessionAirdResource = new UIResource(designerProject, "/", REPRESENTATION_MODEL);
+        localSession = designerPerspective.openSessionFromFile(sessionAirdResource);
+        editor = (SWTBotSiriusDiagramEditor) openRepresentation(localSession.getOpenedSession(), REPRESENTATION_DESCRIPTION_NAME, REPRESENTATION_NAME, DDiagram.class);
+    }
+
+    private SWTBotGefEditPart getCompartment(SWTBotGefEditPart partBot) {
+        return partBot.children() //
+                .stream() //
+                .filter((childPartBot) -> childPartBot.part() instanceof AbstractDNodeContainerCompartmentEditPart) //
+                .findAny() //
+                .orElseThrow();
+    }
+
+    /**
+     * Test of a D&D that involves modifying a border node (deletion and creation) of the drag&dropped element.
+     */
+    public void testDragAndDropWithUpdateNode() {
+        // check the diagram before D&D: diagram with empty container pkg2 and with node cls2
+        List<? extends GraphicalEditPart> diagramContentBefore = editor.getDiagramEditPart().getChildren();
+        assertEquals("Expect only cls2 and pkg2 on the diagram before D&D", 2, diagramContentBefore.size());
+
+        SWTBotGefEditPart cls2Part = editor.getEditPart("cls2", DNodeEditPart.class);
+        SWTBotGefEditPart pkg2Part = editor.getEditPart("pkg2", DNodeContainerEditPart.class);
+
+        List<SWTBotGefEditPart> pkg2ContentBefore = getCompartment(pkg2Part).children();
+        assertTrue("Expect pkg2 empty before D&D", pkg2ContentBefore.isEmpty());
+
+        Point sourceDragLocation = editor.getAbsoluteBounds(cls2Part).getCenter();
+        Point targetDropLocation = editor.getAbsoluteBounds(pkg2Part).getCenter();
+
+        // start log listener
+        startToListenErrorLog(false, true);
+
+        // perform drag cls2 to pkg2
+        editor.drag(sourceDragLocation, targetDropLocation);
+
+        // check the diagram after D&D: diagram with container pkg2 and with node cls2 inside
+        List<? extends GraphicalEditPart> diagramContentAfter = editor.getDiagramEditPart().getChildren();
+        assertEquals("Expect only pkg2 on the diagram after D&D", 1, diagramContentAfter.size());
+        List<SWTBotGefEditPart> pkg2ContentAfter = getCompartment(pkg2Part).children();
+        assertEquals("Expect cls2 in pkg2 after D&D", 1, pkg2ContentAfter.size());
+
+        // in the teardown, check the error log
+    }
```

Comment:
It looks like a comment to TODO work :)
I would prefer
// The error log is checked in the tearDown

Reply:
done